### PR TITLE
Keithley

### PR DESCRIPTION
**Pull request by ichristen**
Keithley 2400 Driver based on Lorenzo's class. Reworked into more of a CC-style Driver. Debugged and working.

### CODE REVIEW

File: Modules/+Drivers/Keithley2400.m [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            end
+            fprintf(obj.VisaHandle,msg);
+            out = fscanf(obj.VisaHandle);
+            if CloseOnDone
+                fclose(obj.VisaHandle);
+            end
+        end
+        function init(obj)
+            obj.output =    obj.get_output();
+%             obj.mode =      obj.get_mode();
```
Comment (ebersin) [inline Modules/+Drivers/Keithley2400.m:95]: Do we want this commented out?

File: Modules/+Drivers/Keithley2400.m [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            output = str2double(obj.query('outp:stat?'));
+            obj.output = output;
+        end
+        
+        % Mode
+        function val = set_mode(obj, val, ~)
+            assert(strcmp(val, 'VOLT') || strcmp(val, 'CURR'), 'mode must be: ''VOLT'' or ''CURR''');
+            obj.cmd(['sour:func:mode ' val]);
+        end
+%         function mode = get_mode(obj)
```
Comment (ebersin) [inline Modules/+Drivers/Keithley2400.m:118]: Should this be implemented, or removed?

File: Modules/+Drivers/Keithley2400.m [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        function val = set_current(obj, val, ~)
+            obj.cmd(['sour:curr:lev:imm:ampl ' num2str(val)]);
+        end
+        function current = get_current(obj)
+            current = str2double(obj.query('sour:curr?'));
+        end
+        
+        % Measure voltage
+        function voltage = measureVoltage(obj)
+            voltage = str2double(strtok(obj.query('meas:VOLT:DC?'),','));  % returned measuremnet consists of 5 numbers, the first is voltage
```
Comment (ebersin) [inline Modules/+Drivers/Keithley2400.m]: `measuremnet` --> `measurement`

Comment (ichristen) [reply Modules/+Drivers/Keithley2400.m]: I blame Lorenzo.

File: Modules/+Drivers/Keithley2400.m (hunk 2 above)
Comment (ichristen) [reply Modules/+Drivers/Keithley2400.m:118]: It should be implemented, but it isn't a priority now.

**Review by mmsutula — APPROVED** [review]

**Pull request merged**